File: Java/Hardware.java
// derived class Hardware - intermediary
class Hardware extends Product{

	/* atribut private */
	private String brand;
	private String model;

	/* konstruktor */
	public Hardware(){}

	public Hardware(String brand, String model){
		this.brand = brand;
		this.model = model;
	}

	/* setter dan getter untuk atribut brand */
	public String getBrand(){
		return brand;
	}

	public void setBrand(String brand){
		this.brand = brand;
	}

	/* setter dan getter untuk atribut model */
	public String getModel(){
		return model;
	}

	public void setModel(String model){
		this.model = model;
	}

	/* method untuk mencetak output atribut dari class Hardware*/
	public void printHardware(){
		System.out.println("Merk           : " + this.getBrand());
		System.out.println("Model          : " + this.getModel());
	}
}

File: Python/Hardware.py
# import class
from Product import Product

class Hardware(Product):
	
	# atribut private
	__brand = ""
	__model = ""

	# konstruktor
	def __init__(self):
		self.__brand = ""
		self.__model = ""

	# setter dan getter untuk atribut brand
	def setBrand(self, brand):
		self.__brand = brand

	def getBrand(self):
		return self.__brand

	# setter dan getter untuk atribut model
	def setModel(self, model):
		self.__model = model
	
	def getModel(self):
		return self.__model;

	# method untuk mencetak output atribut dari class Hardware
	def printHardware(self):
		print("Merk           : " + self.getBrand())
		print("Model          : " + self.getModel())